deployment/ccip/changeset/cs_promote_candidate_test.go

```diff
@@ -0,0 +1,142 @@
+package changeset
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/internal"
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+func Test_PromoteCandidate(t *testing.T) {
+	for _, tc := range []struct {
+		name        string
+		mcmsEnabled bool
+	}{
+		{
+			name:        "MCMS enabled",
+			mcmsEnabled: true,
+		},
+		{
+			name:        "MCMS disabled",
+			mcmsEnabled: false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := testcontext.Get(t)
+			tenv := NewMemoryEnvironment(t,
+				WithChains(2),
+				WithNodes(4))
+			state, err := LoadOnchainState(tenv.Env)
+			require.NoError(t, err)
+
+			// Deploy to all chains.
+			allChains := maps.Keys(tenv.Env.Chains)
+			source := allChains[0]
+			dest := allChains[1]
+			t.Logf("source: %d, dest: %d, home chain: %d, feed chain: %d", source, dest, tenv.HomeChainSel, tenv.FeedChainSel)
+			newAddresses := deployment.NewMemoryAddressBook()
+			err = deployPrerequisiteChainContracts(tenv.Env, newAddresses, allChains)
```

Comment:
should use the cs to set this up

Reply:
Ah yeah we have a ticket for this - this is just copypasta, unless you have a snippet I can copy?

Reply:
example https://github.com/smartcontractkit/chainlink/blob/91192bbea8cfbba2093f70900ff37c1519b1cb05/deployment/ccip/changeset/test_environment.go#L321

Reply:
I wonder if this is already being done then, and this is just redundant.

Reply:
Don't think this is even needed, deleting these lines